test/tech/v3/dataset/mmap_string_test.clj

```diff
@@ -23,31 +23,62 @@
                        :max-chars-per-column 10000000
                        :parser-fn {:abstract :mmap-string} })
 
-        )))
+        ))
+
+  (last (:abstract ds))
+
+
+  (def positions (atom []))
+  (def ds
+    (-> "./test/data/medical-text.csv"
+        (ds/->dataset {:key-fn keyword
+                       :column-whitelist ["cord_uid" "abstract"]
+                       :max-chars-per-column 10000000
+                       :column-opts {:abstract {:mmap-file "/tmp/out.mmap" :positions positions}}
+                       :parser-fn {:abstract :mmap-string} })
+
+        ))
+
+  (.positions
+   (.data
+    (:abstract ds)))
+
+  )
 
 
 (comment
 
-  ;; (def files
-  ;;   (->> (io/file  "/home/carsten/Dropbox/sources/analyseOpinions/corpus")
-  ;;        (file-seq)
-  ;;        (filter #(.isFile %))
-  ;;        (map #(.getPath %))
-  ;;        ))
-
-  ;; (defn file->ds [fpath]
-  ;;   (let [text
-  ;;         (slurp fpath)
-  ;;         ds (->
-  ;;             (ds/->dataset {:text-pointers [text]}
-  ;;                           {
-  ;;                            :parser-fn {:text-pointers [:object #(str->mmap % mmap-file-name )]}})
-  ;;             (add-mmap-text-column  :text-pointers :text))]
-  ;;     ds
-  ;;     ))
-
-  ;; (def all
-  ;;   (apply ds/concat
-  ;;          (map file->ds files)))
+  (def column-opts
+    {:text {:mmap-file "/tmp/corpus.mmap" :positions (atom [])}
+     })
+
+  (def files
+    (->> (io/file  "/home/carsten/Dropbox/sources/analyseOpinions/corpus")
+         (file-seq)
+         (filter #(.isFile %))
+         (map #(.getPath %))
+         ))
+
+  (defn file->ds [fpath]
+    (let [text
+          (slurp fpath)
+          ds (->
+              (ds/->dataset {:text [text]}
+                            { :column-opts column-opts
```

Comment:
I introduced here column-options to configure the MmapStringList.
The needed change looks rather clean.

Maybe we could merge later the `parser-fn` map with  `column-opts` map,
but this would be a breaking change.